### PR TITLE
Ekii/ent1904

### PR DESCRIPTION
**Pull request by erinakii**
Linked to Ticket ENT1904. https://openedx.atlassian.net/browse/ENT-1904. 

Accomplished several things: 

1. There is a single page that lists all programs you are enrolled in, regardless of school.
     ##This is now in the base url '/'. 
2. If you are only enrolled in a single program, it will redirect you to that program.
   ##This is now in '/analytics'
3. This page will be used by the header dropdown (my masters degree link i the header)
   ##is used in the header dropdown or if the user navigates to the hostname

The data is currently hardcoded in the components state. In order to view the logic that redirects the user to '/analytics', you have to delete one of the objects in the array stored in this.state.masters.

I am assuming this shouldn't be merged until the backend API is created? 

Addition by cpappas:
- this now also includes ENT-2008

### CODE REVIEW

File: src/components/MastersTable/MastersTable.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import { navigate } from 'gatsby';
+
+class MastersTable extends Component {
```
[the remark this answers is not in the document]
Comment (erinakii) [reply src/components/MastersTable/MastersTable.jsx]: Thanks! Changed to ProgramsTable

File: src/components/MastersTable/MastersTable.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  componentDidMount() {
+    if (this.state.masters.length < 2) navigate(`${this.state.masters[0].url}`);
+  }
+
+  render() {
+    return (
+      <div className="container">
+        <h1> Masters Degree List </h1>
```
[the remark this answers is not in the document]
Comment (erinakii) [reply src/components/MastersTable/MastersTable.jsx]: Sounds good! Removed spacing and changed to Programs List

**Comment by erinakii** [comment]
@adamstankiewicz @christopappas This branch now includes merge from Chris's ticket 2008 

https://openedx.atlassian.net/browse/ENT-2008

I think that this should be all set now. Let me know if there are other comments you may have

File: src/components/ProgramsTable/ProgramsTable.jsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      .filter(program => program.node.context.programUUID !== null)
+      .map(program => (
+        {
+          uuid: program.node.context.programUUID,
+          slug: program.node.context.programSlug,
+          name: program.node.context.programName,
+        }
+      ));
+  }
+  state = {
```
[the remark this answers is not in the document]
Comment (erinakii) [reply src/components/ProgramsTable/ProgramsTable.jsx]: TY!

File: src/components/ProgramsTable/ProgramsTable.jsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        uuid: '11111111-bbbb-cccc-dddd-eeeeeeeeeeee',
+      },
+      {
+        slug: 'missingprogram2',
+        uuid: '22222222-bbbb-cccc-dddd-eeeeeeeeeeee',
+      },
+    ];
+
+    this.validateUserPrograms(enrolledPrograms);
+    if (this.state.hasValidProgram === true && this.programData.length === 1) navigate(`${this.programData[0].slug}`);
```
[the remark this answers is not in the document]
Comment (erinakii) [reply src/components/ProgramsTable/ProgramsTable.jsx]: this is outdated so not an issue anymore I believe

File: src/components/ProgramsTable/ProgramsTable.jsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+          <h1>Program List</h1>
+          <div className="table-responsive">
+            <table className={classNames('table', 'table-sm', 'table-striped')}>
+              <thead>
+                <tr>
+                  <th>Program</th>
+                </tr>
+              </thead>
+              <tbody>
+                {
```
[the remark this answers is not in the document]
Comment (erinakii) [reply src/components/ProgramsTable/ProgramsTable.jsx]: done!

File: src/components/Layout/Layout.jsx [hunk 6]
```diff
@@ -4,6 +4,8 @@ import PropTypes from 'prop-types';
 import SiteHeader from '@edx/frontend-component-site-header';
 import SiteFooter from '@edx/frontend-component-footer';
 import { connect } from 'react-redux';
+// import { IntlProvider } from 'react-intl';
```
Comment (christopappas) [inline src/components/Layout/Layout.jsx]: we should remove this comment

**Pull request merged**